src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(5)) {
+            case 0 :
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextDouble());
+            case 1 :
+                return new Circle(colorSupplier.getRandomColor(),
+                        random.nextDouble());
+            case 2 :
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextDouble(),
+                        random.nextDouble());
+            case 3 :
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextDouble(),
+                        random.nextDouble(),
+                        random.nextDouble());
+            case 4 :
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextDouble(),
+                        random.nextDouble());
```

Comment:
Creating figures with expressions directly in the constructor reduces readability. It's better to create separate variables for the dimensions and pass them to the constructor.